File: src/main/java/MessageResource.java
import models.Message;
import services.MessageService;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

@Path("/messages")
public class MessageResource {

    MessageService messageService = new MessageService();

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public List<Message> getMessages() {
        return messageService.getMessages();
    }

    @GET
    @Path("/{messageId}")
    @Produces(MediaType.APPLICATION_JSON)
    public Message test(@PathParam("messageId") long messageId) {
        return messageService.getMessage(messageId);
    }
}

File: src/main/java/resources/MessageResource.kt
package resources

import models.Message
import services.MessageService
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/messages")
public class MessageResource {

    var messageService = MessageService()


    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun getMessages(): List<Message> {
        return messageService.getMessages()
    }

    @GET
    @Path("/{messageId}")
    @Produces(MediaType.APPLICATION_JSON)
    fun getMessageById(@PathParam("messageId") messageId: Long): Message? {
        return messageService.getMessage(messageId)
    }
}